lib/checkunusedfunctions.cpp

```diff
@@ -235,6 +235,49 @@ void CheckUnusedFunctions::parseTokens(const Tokenizer &tokenizer, const char Fi
 }
 
 
+static bool isOperatorFunction(const std::string & funcName) {
+    /* Operator functions are invalid function names for C, so no need to check
+     * this in here. As result the returned error function might be incorrect.
+     *
+     * List of valid operators can be found at:
+     * http://en.cppreference.com/w/cpp/language/operators
+     *
+     * Conversion functions must be a member function (at least for gcc), so no
+     * need to cover them for unused functions
+     *
+     * Literal functions are treated at a different place, so no check is done
+     * in here
+     */
+    const std::string operatorPrefix = "operator";
+    if (funcName.compare(0, operatorPrefix.length(), operatorPrefix) != 0) {
+        return false;
+    }
+
+    std::string opName = funcName.substr(operatorPrefix.length());
+
+    /* Operation overload + allocation/deallocation
+     *
+     * Don't treat the following, as they have to be a member function:
+     * '=', '()', '[]', '->', '->*'
+     */
+    const std::vector<std::string> knownOperators = {
+        "+", "-", "*", "/", "%",  "&", "|", "~", "!", "<", ">", "+=", "-=",
+        "*=", "/=", "%=", "^=", "&=", "|=", "<<", ">>", ">>=", "<<=", "==",
+        "!=", "<=", ">=", "&&", "||", "++", "--", ",",
+        // Allocation variants; space is delete within the simplification
+        "new", "new[]",
+        // Deallocation variants; space is delete within the simplification
+        "delete", "delete[]"
+    };
+    for (auto const & curOp : knownOperators) {
```

Comment:
please don't use a ranged for loop. That is C++11. It seems clumpsy to check for every known operator. If opName is any operator (!std::isalnum(opName[0] && opName[0] != '_')) then I think this function could return true.